### DISCUSSION TITLE
Multi-Select Picker: default values via data provider possible?

### DISCUSSION THREAD

**Discussion by pandel** [Budibase/budibase#7277] [Help] [answered]
Hi!

I have the following structure:

Data Provider A -> multi-select field values
Data Provider B -> table with data

I set up a filter for 'Data Provider B' with the values from the multi-select field, so that the table is updated accordingly.
This is working totally fine.

BUT:
ATM, when I load the page, the default value for the multi-picker is empty and so nothing is shown in the table, unless I select some options in the multi-select picker, obviously.

Now, what I want is, that the default value of the multi-picker is like having everything selected right at page load, but I don't know how to write a JavaScript or Handlebar formula so that the multi-picker gets its default values set accordingly.

I tried something like this as the default value, but it doesn't work:

```
const select_values = $("Data Provider A.Rows");
let s = [];

for (let r in select_values) {
    s.push(r.ColumnName);
}

return s;
```

If I set the default value like the following example it works (but this list is going to change and grow over time and I'd really like to get the values directly from the data provider:

`return ['valA', 'valB', 'valC']`

Does anybody know how to accomplish this?

Regards,
Holger

**Comment by melohagan** [comment]
Hi @pandel 

There is an issue open that might be related to this: https://github.com/Budibase/budibase/issues/5765

Would you be able to share your app export to support@budibase.com please? Then I can investigate further.

**Reply by pandel** [reply]
@melohagan
Yes, I think I can send you the export, but the datasource is MySQL and I cannot share this as easy as the export. Is this a problem?

**Reply by melohagan** [reply]
It should be okay. I just want to see how you're using the binding. I can plugin a dummy table to test it.

**Reply by pandel** [reply]
I sent you the export. Many thanks in advance!

**Reply by melohagan** [reply]
Which screen am I to look at, and is your binding code for the default value still there?

**Reply by pandel** [reply]
You have to go to screen "/cert-meldungen" and there in "Category filter" multi-select. The current code is the the hard-coded (working) one: `return ["kritisch","hoch","mittel","niedrig"]`.

I actually tried:
```
const select_values = $("Data Provider Cats.Rows");
let s = [];

for (let r in select_values) {
    s.push(r.Kategorie);
}
return s;
```

The table "Kategorien", used in `Data Provider Cats`, has been created via Budibase and returns only one column "Kategorie". But there must be an "id" column, too, because Budibase creates them automatically, I think...

**Comment by melohagan** [comment, marked as answer]
Hey @pandel 

I should have spotted it. You need to use **of** instead of **in**. JavaScript madness! 

```js
const select_values = $("Data Provider Cat.Rows");
let s = [];

for (let r of select_values) {
    s.push(r.Kategorie);
}
return s;
```

Also it should be Data Provider Cat (not Cats). Remove the binding and re-add it by clicking the item in the left hand panel to be sure.

**Reply by pandel** [reply]
Although "Cats" was a typo, you found it! It is "of", not "in"! It is working now.

Many many thanks for taking you time!
